src/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs

```diff
@@ -821,8 +821,7 @@ public override unsafe int GetCharCount(byte* bytes, int count, DecoderNLS decod
             byte[] byteBuffer = new byte[1];
             char* charEnd = chars + charCount;
 
-            DecoderFallbackBufferHelper fallbackHelper = new DecoderFallbackBufferHelper(
-                decoder != null ? decoder.FallbackBuffer : DecoderFallback.CreateFallbackBuffer());
```

Comment:
@layomia, @tarekgh, @krwq should take a look.

Also, are there tests covering all of these paths and that would catch any issues here?

Reply:
Although we are testing the encoders and decoders with Utf8 but these tests are not covering SBCS encodings. We should add a test for this change to cover the fallback actions with SBCS encoding. the test should be same idea that @layomia was doing but we need to have the test use the default the encoders/decoders. @slozier please let me know if you  need help writing this test.